app/models/claim/verify_response_parameters_parser.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+class Claim
+  # Used to convert the parameters from a successful Verify::Response into an
+  # attributes Hash that can be used to update a claim. For example,
+  #
+  #   Claim::VerifyResponseParametersParser.new(verify_response.parameters).attributes
+  #   => { full_name: "Margaret Elaine Heafield Hamilton", date_of_birth: "1936-08-17", ...}
+  #
+  # As well as keys for the personal information of the user (name, address,
+  # etc) the returned Hash includes two additional keys:
+  #
+  #   verified_fields – these are the keys for the attributes that have come
+  #                     back in the Verify response. Recording these allows us
+  #                     to determine which attributes came from Verify and
+  #                     therefore should not be editable by the user.
+  #
+  #   verify_response – this is the entire Verify response payload. We record
+  #                     this so that we can examine the original payload to
+  #                     debug issues in the way we parse responses, but also so
+  #                     we can retrospectively re-parse responses in future if
+  #                     necessary.
+  #
+  class VerifyResponseParametersParser
+    class MissingResponseAttribute < StandardError; end
+
+    def initialize(response_parameters)
+      @response_parameters = response_parameters
+    end
+
+    def attributes
+      identity_attributes.merge(verified_fields: verified_fields, verify_response: @response_parameters)
+    end
+
```

Comment:
Should the methods below here be private too? (given they're never called outside the class)

Reply:
Keeping them public made it easier to write tests for the specific nuances for parsing each of the attributes. If they were made private, the tests would need to be adjusted so that they were all against the `attributes` method. Probably wouldn't make things that much more complicated though to be fair...

Reply:
I'm going to suggest we wait until @robbiepaul has done the work to split out the names as otherwise the specs will need significant rewrite to test against `:full_name` and then have to be un-rewritten to test against the individual components later.